### PR TITLE
improv: don't store profiling in files #192

### PR DESCRIPTION
**Pull request by Eywek**
Not ready to merge: We need to send packet to profiling channel instead of axm:reply

### CODE REVIEW

File: src/profiling/profilingHeapFallback.ts [hunk 1]
```diff
@@ -70,7 +69,7 @@ export default class ProfilingHeapFallback implements ProfilingType {
         }
       )
     }).then((buffer) => {
-      return FileUtils.writeDumpFile(buffer, '.heapprofile')
```
Comment (wallet77) [inline src/profiling/profilingHeapFallback.ts]: Maybe we can remove file FileUtils if it's not use anymore.
Or just remove some methods.

Comment (Eywek) [reply src/profiling/profilingHeapFallback.ts]: FileUtils is used for coverage

Comment (wallet77) [reply src/profiling/profilingHeapFallback.ts]: Yes but it works the same way as Heap and CPU profiling so we should change it also.

**Comment by Eywek** [comment]
We need this https://github.com/keymetrics/km-api/pull/799

File: src/profiling/profilingCPU.ts [hunk 2]
```diff
@@ -87,7 +86,7 @@ export default class ProfilingCPU implements ProfilingType {
           data.timeDeltas = []
         }
 
-        return resolve(FileUtils.writeDumpFile(JSON.stringify({
+        return resolve(JSON.stringify({
```
Comment (vmarchaud) [inline src/profiling/profilingCPU.ts]: We don't need to stringify here, it should be done at transport level

File: src/profiling/profilingCPUFallback.ts [hunk 3]
```diff
@@ -54,6 +53,6 @@ export default class ProfilingCPUFallback implements ProfilingType {
   private getProfileInfo () {
     const cpu = this.profiler.stopProfiling(this.nsCpuProfiling)
 
-    return FileUtils.writeDumpFile(JSON.stringify(cpu))
+    return JSON.stringify(cpu)
```
Comment (vmarchaud) [inline src/profiling/profilingCPUFallback.ts]: We don't need to stringify here, it should be done at transport level

File: src/profiling/profilingHeap.ts [hunk 4]
```diff
@@ -66,7 +65,7 @@ export default class ProfilingHeap implements ProfilingType {
         return reject(err)
       }
 
-      resolve(FileUtils.writeDumpFile(JSON.stringify(data.profile), '.heapprofile'))
+      resolve(JSON.stringify(data.profile))
```
Comment (vmarchaud) [inline src/profiling/profilingHeap.ts]: We don't need to stringify here, it should be done at transport level

File: src/profiling/profilingCPU.ts (hunk 2 above)
Comment (Eywek) [reply src/profiling/profilingCPU.ts]: The file content should be a string, not an object

File: src/features/actions.ts [hunk 5]
```diff
@@ -48,7 +48,7 @@ export default class ActionsFeature implements Feature {
     if (actionData) {
       // In case 2 arguments has been set but no options has been transmitted
       if (fn.length === 2) {
-        return fn(data.opts || {}, reply)
+        return fn(typeof data === 'object' ? data.opts : {}, reply)
```
Comment (vmarchaud) [inline src/features/actions.ts]: we need to check both `data` and `opts`

**Pull request merged**